src/core/externalstorage/qgsexternalstorageregistry.cpp

```diff
@@ -0,0 +1,49 @@
+/***************************************************************************
+  qgsexternalstorageregistry.cpp
+  --------------------------------------
+  Date                 : March 2021
+  Copyright            : (C) 2021 by Julien Cabieces
+  Email                : julien dot cabieces at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsexternalstorageregistry.h"
+
+#include "qgsexternalstorage.h"
+#include "qgssimplecopyexternalstorage_p.h"
+
+QgsExternalStorageRegistry::QgsExternalStorageRegistry()
+{
+  registerExternalStorage( new QgsSimpleCopyExternalStorage() );
+}
+
+QgsExternalStorageRegistry::~QgsExternalStorageRegistry()
+{
+  qDeleteAll( mBackends );
+}
+
+QgsExternalStorage *QgsExternalStorageRegistry::externalStorageFromType( const QString &type )
+{
+  return mBackends.value( type );
+}
+
+QList<QgsExternalStorage *> QgsExternalStorageRegistry::externalStorages() const
+{
+  return mBackends.values();
+}
+
+void QgsExternalStorageRegistry::registerExternalStorage( QgsExternalStorage *storage )
+{
+  mBackends.insert( storage->type(), storage );
+}
+
+void QgsExternalStorageRegistry::unregisterExternalStorage( QgsExternalStorage *storage )
+{
+  delete mBackends.take( storage->type() );
```

Comment:
`reset()` instead of `delete` if we use a `unique_ptr`

Reply:
BTW, what do you think about passing the storage type in parameter instead of the pointer (considering that it's not really used)?

Reply:
I would prefer to stay consistent with other registry class like [here](https://github.com/qgis/QGIS/blob/master/src/core/gps/qgsgpsconnectionregistry.h#L54) or [there](https://github.com/qgis/QGIS/blob/master/src/core/project/qgsprojectstorageregistry.h#L52), but I have no strong opinion.